trillian/integration/copier.go

```diff
@@ -79,16 +78,16 @@ func NewCopyChainGeneratorFromOpts(ctx context.Context, client *client.LogClient
 	var start, limit time.Time
 	var err error
 	if cfg.NotAfterStart != nil {
-		start, err = ptypes.Timestamp(cfg.NotAfterStart)
-		if err != nil {
+		if err := cfg.NotAfterStart.CheckValid(); err != nil {
```

Comment:
How about caching `cfg.NotAfterStart` to a variable in the `if` condition, and reusing it here and 3 lines below?

Same in other similar places in this PR.

Reply:
I'm trying to keep this PR mostly mechanical, to avoid little incidents like shadowing a variable where one should have been assigned to, for example.